File: BE/VibeTunes.Domain/Entities/Album.cs
using System.ComponentModel.DataAnnotations.Schema;
using VibeTunes.Domain.Common;

namespace VibeTunes.Domain.Entities;

public sealed class Album : BaseEntity
{
    public Guid ArtistId { get; set; }
    [ForeignKey("ArtistId")]
    public Artist Artist { get; set; }
    
    public string Title {get; set;}
    public DateOnly ReleaseDate {get; set;}
    public int Streams { get; set; } = 0;
    public string CoverImgUrl {get; set;}
    public bool IsReleased { get; set; } = false;
    
    public ICollection<Song> SongsList { get; set; } = new List<Song>();
}

File: FE/types/album.ts
import {Song} from "@/types/song";

export interface Album {
    id: string,
    artistId: string,
    artist: string,
    title: string,
    releaseDate: string,
    streams: number,
    coverImgUrl: string,
    isReleased: boolean,
    songsList: Song[]
}